data/trie/trieStorageManager.go

```diff
@@ -487,10 +509,19 @@ func (tsm *trieStorageManager) removeSnapshot() {
 	removeSnapshot(snapshot, removePath)
 }
 
-func removeSnapshot(db data.DBWriteCacher, path string) {
+func disconnectSnapshot(db data.DBWriteCacher) error {
 	err := db.Close()
 	if err != nil {
-		log.Error("trie storage manager: removeSnapshot", "error", err.Error())
+		log.Error("trie storage manager: disconnectSnapshot", "error", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func removeSnapshot(db data.DBWriteCacher, path string) {
+	err := disconnectSnapshot(db)
+	if err != nil {
```

Comment:
You should only return the error from disconnectSnapshot() without logging it, and log it here.

Reply:
right

Reply:
Done!